ci-operator/step-registry/ibmcloud/provision/iamservice/minimal-permission/ibmcloud-provision-iamservice-minimal-permission-commands.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+#pre create the iam service-id and based it create the iam service api key for ibmcloud-ipi-minimal-permission test
+# based on doc https://docs.openshift.com/container-platform/4.14/installing/installing_ibm_cloud_public/installing-ibm-cloud-account.html#required-access-policies-ibm-cloud_installing-ibm-cloud-account
+
+function run_command() {
+    local CMD="$1"
+    echo "Running Command: ${CMD}"
+    eval "${CMD}"
+}
+
+# IBM Cloud CLI login
+function ibmcloud_login {
+    export IBMCLOUD_CLI=ibmcloud
+    export IBMCLOUD_HOME=/output
+    region="${LEASED_RESOURCE}"
+    export region
+    echo "Try to login..."
+    "${IBMCLOUD_CLI}" config --check-version=false
+    "${IBMCLOUD_CLI}" login -r ${region} --apikey @"${CLUSTER_PROFILE_DIR}/ibmcloud-api-key"
+}
+
+#####################################
+##############Initialize#############
+#####################################
+ibmcloud_login
+
+CLUSTER_NAME="${NAMESPACE}-${UNIQUE_HASH}"
+serviceName="${CLUSTER_NAME}-perm"
+keyName="${serviceName}-key"
+keyFile="${ARTIFACT_DIR}/${CLUSTER_NAME}-key.json"
+
+## create the server id
+run_command "${IBMCLOUD_CLI} iam service-id-create ${serviceName}"
+run_command "${IBMCLOUD_CLI} iam service-id ${serviceName}"
+
+
+## create the service key
+run_command "${IBMCLOUD_CLI} iam service-api-key-create ${keyName} ${serviceName} -d 'the key for minimal permission test in ${CLUSTER_NAME}' --file ${keyFile}"
+
+if [[ ! -f "${keyFile}" ]]; then
+  echo "Fail to find the file which saved the service api key, abort." && exit 1
+fi
+echo "Reading api key info from ${keyFile}..."
+run_command "cat ${keyFile}"
+newKey=$(cat ${keyFile} | jq -r .apikey)
+
+echo "create the required access policies based on the required-access-policies-ibm-cloud_installing-ibm-cloud-account doc and assign them to the service ID ${serviceName} ..."
+
+$IBMCLOUD_CLI iam service-policy-create ${serviceName} --roles "Viewer,Operator,Editor,Administrator,Reader,Writer,Manager" --service-name dns-svcs
```

Comment:
maybe we can consider using a function to encapsulate the command 
`$IBMCLOUD_CLI iam service-policy-create ${serviceName} --roles ${roles} --service-name ${service_name}` to avoid writing it so many times.